### ISSUE TITLE
Needed developments for science tools

### ISSUE THREAD

**Issue by jjlk** [gammapy/gammapy#808]
Hi @cdeil , @joleroi , @registerrier , @bkhelifi , @adonath , @TarekHC , 
In order for gammapy to be able to handle science use cases in CTA I list below some needed features:  
- [ ]  Make a class to store CTA performance from public IRF (effective area, background rate, energy dispersion and PSF)
- [ ]  Make a new class which inherits from `SpectralModel` to be able to handle an absorbed model  (e.g. EBL)
- [ ]  Adapt `SpectrumSimulation` class to handle background from IRF, see first point

For the first point, I think we should focus on the public IRF available [here](https://portal.cta-observatory.org/Pages/CTA-Performance.aspx). So I propose to create a new class, `CTAPerf`, that we will fill from the ascii files. In addition we will need a new class to handle the background rate (`BgRate`?), for convenience. We will also need to add the CTA performance files in gammpy-extra, right?

For the second point, I'm convinced, by @registerrier, that widely-used binary format is needed for the absorption models. I think the one comming from the X-ray community would be the best. It's FITS and there are ftools that can produce them very easily from ASCII files ([flx2tab](https://heasarc.gsfc.nasa.gov/ftools/caldb/help/flx2tab.html)). So I guess we could add a machinery in gammapy to handle thoses files. Meaning that given a parameter and an energy, the thing will output an absorption value. I discussed with @cdeil about absorbed spectral model and at the end we should have something like that:
```
class AbsorbedSpectralModel:
   def __init__(spectral_model, ebl_model):
        self.spectral_model = spectral_model
        self.ebl_model = ebl_model
        self.parameters = {}
    def evaluate(self, energy):
         flux = self.spectral_model.evaluate(energy)
          absorption = self.ebl_model.evaluate(energy)
        return flux * absorption 
```

For the third point, I guess it shouldn't be too difficult to adapt the `SpectrumSimulation` class to deal with background comming from IRF (rate in Hz). Maybe we could discuss about that @joleroi ?

I'm going to take the tasks in order, if anyone has any suggestions, vetos or comments, you are very welcome. 

I propose to keep the issue open until everything is done.

**Comment by cdeil** [comment]
@jjlk - Sounds great!

It's usually best to do one issue per point, because otherwise discussion on multiple topics is in one linear thread that is hard to follow. Still, here's some quick comments:

---

For the performance you already added [gammapy.scripts.CTAIrf](http://docs.gammapy.org/en/latest/api/gammapy.scripts.CTAIrf.html), which access FITS prod 2 IRFs from CTA. Did you mean to extend this or also some other IRF files from CTA?

Then we need a class or tools to compute sensitivity. Maybe best to make it separate from the CTAIrf class, but re-use it as an input? Feel free to make an issue or pull request (or example IPython notebook) with a proposal, or let me know if you'd like me to propose a class / API.

Note that @woodmd has been developing code and possibly a format for sensitivity in Fermipy (see [here](http://fermipy.readthedocs.io/en/latest/advanced/sensitivity.html) and [here](https://github.com/fermiPy/fermipy/blob/master/fermipy/scripts/flux_sensitivity.py). 

There's also multiple scripts and format in HESS (see e.g. [here](https://hess-confluence.desy.de/confluence/display/HESS/Task%3A+H.E.S.S.+II+Sensitivity+Tools)) concerning sensitivity and observation time inputs / outputs.

The ideal long-term solution would be to define a format for the inputs / outputs as a separate page at http://gamma-astro-data-formats.readthedocs.io/en/latest/results/index.html, and then implement support for that in Gammapy, by long-term moving over SED code from Fermipy to Gammapy.

@jjlk - What this means for you, now isn't clear to me. Just add whatever you need to get your job done. And maybe if you have time, split out the documentation of inputs / outputs of the sensitivity computation into a docs page in the specs, instead of a docs page in Gammapy?

For now, yes, all test files should also go in `gammapy-extra`, so that they are available from the Gammapy continuous integration tests and docs build.

---

For EBL absorption it's the same. OK to add to Gammapy whatever you find useful.

If you have the time, again, defining a format separately in the `gamma-astro-data-formats` repo would be a good step towards avoiding code and EBL absorption file duplication from the various ST codes.

I don't know much about EBL absorption. If the files from the various models can be brought into the same format, great! If not, the still having *any* format and reference files described in `gamma-astro-data-formats` would be great IMO, so that it's usable by Gammapy and anyone else that wants to.

If you don't want to be bogged down in spec and format discussions, you can also just start adding and using stuff in Gammapy, and then the spec can come later.

---

In principle `SpectrumSimulation` already can handle background. Of course, if there's some feature missing, or it's not working with your background files, yes, let's add it. Talk to @joleroi about that.

**Comment by cdeil** [comment]
I want to release Gammapy 0.6 tomorrow. Moving this to the 0.7 milestone. Note that if we make progress in the coming weeks, 0.7 can be very soon (e.g. before the 1DC data release).

**Comment by cdeil** [comment]
@jjlk - What are the action items here? Can this old issue be closed? If no - could you please summarise what could / should be done?

Moving to v0.8 milestone for now.

**Comment by cdeil** [comment]
I think most of these features are available now.
Please open specific new issues for things that should be added / changed / removed.
